src/gluonts/dataset/common.py

```diff
@@ -207,13 +213,144 @@ def files(self) -> List[Path]:
         List[Path]
             List of the paths of all files composing the dataset.
         """
-        return util.find_files(self.path, self.is_valid)
+        return util.find_files(self.paths, self.is_valid)
 
     @classmethod
     def is_valid(cls, path: Path) -> bool:
-        # TODO: given that we only support json, should we also filter json
-        # TODO: in the extension?
-        return not (path.name.startswith(".") or path.name == "_SUCCESS")
+        valid_endings = [
+            ".json",
+            ".jsonl",
+            ".json.gz",
+            ".jsonl.gz",
+        ]
+        if not any(path.name.endswith(ending) for ending in valid_endings):
+            return False
+        return not path.name.startswith(".")
+
+
+class ArrowDataset(Dataset):
```

Comment:
I think I don't want this to be in `common`, since I want to see `common` to be removed in the future.

Can't we just define it all in `dataset/arrow.py`, and then import the `ArrowDataset` in `dataset/__init__.py`, so that we can then do:

```py
from gluonts.dataset import ArrowDataset
```

Reply:
yes, makes sense